src/main/java/org/mule/tools/devkit/sonar/checks/structure/GitIgnoreExistsCheck.java

```diff
@@ -0,0 +1,80 @@
+package org.mule.tools.devkit.sonar.checks.structure;
+
+import com.google.common.base.Predicates;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import org.apache.commons.io.filefilter.TrueFileFilter;
+import org.apache.commons.io.filefilter.WildcardFileFilter;
+import org.apache.commons.lang.StringUtils;
+import org.apache.maven.project.MavenProject;
+import org.mule.tools.devkit.sonar.checks.ConnectorIssue;
+import org.sonar.api.batch.fs.FileSystem;
+import org.apache.commons.io.FileUtils;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+
+@Rule(key = GitIgnoreExistsCheck.KEY, name = ".gitignore should be present", description = "There should exist a .gitignore file in root folder.", priority = Priority.CRITICAL, tags = { "connector-certification" })
+public class GitIgnoreExistsCheck  extends ExistingResourceCheck {
```

Comment:
It looks like the git hooks that format the code is not active. Please install it and format all code.